### DISCUSSION TITLE
Painting app written in PIL/Pillow and CustomTkinter

### DISCUSSION THREAD

**Discussion by limafresh** [python-pillow/Pillow#8561] [Show and tell]
[`Visit repository on GitHub`](https://github.com/limafresh/Brushshe)

<p align="center">
  <img src="https://raw.githubusercontent.com/limafresh/Brushshe/refs/heads/main/Brushshe/icons/logo.svg" alt="logo" width="100" height="100">
</p>

**Brushshe** is a painting app written in CustomTkinter and Pillow with stickers, shapes, effects and frames. The program's mascot and "helper" is an eagle named Brucklin. The program also has its own gallery.

![Снимок экрана_2025-02-07_09-10-43](https://github.com/user-attachments/assets/41ed37cd-497a-46a4-98c3-f2c5810d5a3b)

<p align="center">Thanks for reading!</p>
<p align="center">If you liked it, leave stars in the repository and comments</p>

**Comment by hugovk** [comment]
Nice, thanks for sharing! :)

**Reply by limafresh** [reply]
Thanks! By the way, I have now translated it into English
⬇️
https://github.com/limafresh/Brushshe

**Comment by limafresh** [comment]
Now there is a portable .exe file that runs Brushshe without installing Python and dependencies. You can download it from the [repository page](https://github.com/limafresh/Brushshe/releases)

**Comment by limafresh** [comment]
Since v1.0.0, Brushshe has switched to PIL.ImageDraw, making it a true raster graphics editor.

**Comment by limafresh** [comment]
## Versions 1.4.0 - 1.5.0
![1](https://github.com/user-attachments/assets/a7f72bd3-27c0-4405-8c52-03b503fdd377)

### What's new?
- Updated UI, it became more convenient and beautiful;
- Improved shape functions, thin canvas.create_[shape name] is used to draw them, and when the mouse is released, the shape is drawn using self.draw.[shape name];
- Added redo function;
- Bugs fixed.

**Comment by limafresh** [comment]
## Versions 1.6.0 - 1.14.3

![screenshot](https://github.com/user-attachments/assets/b0c12df9-1192-4901-ac65-2c08bda9e697)
*Mascot of Brushshe. Drawn 100% in Brushshe. By Chip Vield, licensed under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)*

### What's new?
+ Thickness clearly from 1 to 50
+ Improved quality of shapes and lines, shapes can be drawn from right to left, and there are no sloppy tips in the line
+ Added Bezier curve
+ Improved palette design
+ New set of icons on the toolbar in a single style
+ Fixed a bug with the shape menu
+ A more pleasant color theme
+ Scroll bars
+ Enlarge and reduce the image
+ Delete buttons in the gallery
+ Rotate left-right
+ Resize window where you can change the file size by cropping and scaling

🌟 If you liked the project, [put a star](https://github.com/limafresh/Brushshe). This will help in development.

And feel free to contribute.
